sqlmesh/core/context.py

```diff
@@ -2042,6 +2084,21 @@ def _register_notification_targets(self) -> None:
             event_notifications, user_notification_targets, username=self.config.username
         )
 
+    def _load_factory(
+        self,
+        method: str,
+        loader: t.Optional[Loader] = None,
+        configs: t.Optional[t.Dict[Path, C]] = None,
+        **kwargs: t.Any,
+    ) -> LoadedProject | None:
+        if loader and configs:
```

Comment:
I don't think you can ever have a truthy `loader` and a falsey `configs` here, so the latter may be unnecessary.

Reply:
yes that's true this is only there to pass a mypy check